File: pytools/examples/golang/src/client/proto/ack_scene_players.go
package proto

import (
	"packet"
)

type AckScenePlayers struct {
	players                  []*MsgScenePlayer
}

func AckScenePlayersDecode(pack *packet.Packet) *AckScenePlayers {
	ackScenePlayers := &AckScenePlayers{}

	playersCount := pack.ReadUint16()
	for ;playersCount > 0; playersCount-- {
		ackScenePlayers.players = append(ackScenePlayers.players, MsgScenePlayerDecode(pack))
	}
	return ackScenePlayers
}

func (this *AckScenePlayers) GetPlayers() []*MsgScenePlayer {
	return this.players
}

File: pytools/data/data.code.php.client/AckScenePlayers.php
<?php
namespace protocol;

use network\Packet;


class AckScenePlayers
{
	private $players = [];


	public function __construct(&$packet)
	{
		$this->decode($packet);
	}

	public function decode(&$packet)
	{
		$playersCount = $packet->readU16();
		for ($i = 0; $i < $playersCount; $i++)
		{
			array_push($this->players, new MsgScenePlayer($packet));
		}
	}


	public function getPlayers()
	{
		return $this->players;
	}

}
